### PR TITLE
pallet-evm: return Ok(()) when EVM execution fails

### PR DESCRIPTION
**Pull request by sorpaas**
When the EVM execution fails, it still needs to apply the gas fee to state. As a result, this should return `Ok(())` in terms of `DispatchResult`. This PR fixes the issue.

### CODE REVIEW

**Comment by shawntabrizi** [comment]
A test?

**Comment by sorpaas** [comment]
Added the `fail_call_return_ok` test for this PR. Will work on more tests in separate PRs.

File: frame/evm/src/lib.rs [hunk 1]
```diff
@@ -300,15 +301,24 @@ decl_module! {
...
 				value,
 				gas_limit,
 				gas_price,
 				nonce,
-			).map_err(Into::into)
+			)? {
+				ExitReason::Succeed(_) => {
+					Module::<T>::deposit_event(Event::<T>::Executed(target));
+				},
+				_ => {
```
Comment (thiolliere) [inline frame/evm/src/lib.rs:316]: I'm usually in favor of full match pattern, but seems fine here

**Comment by sorpaas** [comment]
`gitlab-check-polkadot-companion-build` failure seems unrelated to this PR, investigating.

**Pull request merged**